File: Leetcode/204-Count-Primes/CountPrimes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n<3:
            return 0
        primes = [True]*n
        primes[0] = False
        primes[1] = False
        i=2
        while i*i<n:
            if primes[i]:
                j = i+i
                while j<n:
                    primes[j]=False
                    j+=i
            i+=1
        count=0
        for b in primes:
            if b:
                count+=1
        return count
    

File: Leetcode/204-Count-Primes/CountPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n<3) {
            return 0;
        }
        bool primes[n];
        for (int i=2; i<n; i++) {
            primes[i] = true;
        }
        primes[0] = false;
        primes[1] = false;
        for (int i=2; i*i<n; i++) {
            if (primes[i]) {
                for (int j=i+i; j<n; j+=i) {
                    primes[j] = false;
                }
            }
        }
        int count=0;
        for(int i=0; i<n; i++) {
            if (primes[i]) {
                count++;
            }
        }
        return count;
    }
};
